code/datums/station_traits/job_traits.dm

```diff
@@ -93,3 +108,55 @@
 	// monkey carries the crates, the age of robot is over
 	if(GLOB.cargo_ripley)
 		qdel(GLOB.cargo_ripley)
+
+/datum/station_trait/job/bridge_assistant
+	name = "Bridge Assistant"
+	button_desc = "Sign up to become the Bridge Assistant and watch over the Bridge."
+	weight = 2
+	report_message = "We have installed a Bridge Assistant on your station."
+	show_in_report = TRUE
+	can_roll_antag = CAN_ROLL_PROTECTED
+	job_to_add = /datum/job/bridge_assistant
+
+/datum/station_trait/job/bridge_assistant/New()
+	. = ..()
+	RegisterSignal(SSatoms, COMSIG_SUBSYSTEM_POST_INITIALIZE, PROC_REF(add_coffeemaker))
+
+/datum/station_trait/job/bridge_assistant/on_lobby_button_update_overlays(atom/movable/screen/lobby/button/sign_up/lobby_button, list/overlays)
+	. = ..()
+	overlays += "bridge_assistant"
+
+/// Creates a coffeemaker in the bridge, if we don't have one yet.
+/datum/station_trait/job/bridge_assistant/proc/add_coffeemaker(datum/source)
+	SIGNAL_HANDLER
+	var/area/bridge = GLOB.areas_by_type[/area/station/command/bridge]
+	if(isnull(bridge)) //no bridge, what will he assist?
+		return
+	var/list/possible_coffeemaker_positions = list(/area/station/command/bridge, /area/station/command/meeting_room)
+	for(var/possible_position in possible_coffeemaker_positions) //don't spawn a coffeemaker if there is already one on the bridge
+		var/area/possible_area = GLOB.areas_by_type[possible_position]
+		if(possible_area && (locate(/obj/machinery/coffeemaker) in possible_area))
+			return
+	var/list/tables = list()
+	for(var/obj/structure/table/table in bridge)
```

Comment:
No `in world` loops, please

Reply:
what do you mean its in a specific area

Reply:
When you iterate over `object in area` you're actually doing `for object in world, if object in area ...`

Reply:
why didnt you say that instead of making me ask that

Reply:
![image](https://github.com/tgstation/tgstation/assets/51863163/00e95282-d010-495a-8215-20bc2fb213c2)

Reply:
not an answer

Reply:
I sent that before github showed me your second response

I assumed you already knew

Reply:
so what now do i have to loop through every turf for tables

Reply:
```dm
for(var/turf/area_turf as anything in bridge.get_contained_turfs())
    var/obj/structure/table/table = locate() in area_turf
    if(isnull(table))
        continue
    ...
```
I don't actually know if `locate() in area` is any better, I assume it's not, but that's just a guess